src/common/utils/batch.ts

```diff
@@ -0,0 +1,90 @@
+import sizeof from 'object-sizeof';
+import { BatchResult } from '../../steps';
+import { BatchError } from '../errors/batch';
+export class BatchUtils {
+  static chunkArrayBySizeAndLength(
+    array: any[],
+    options: { maxSizeInBytes?: number; maxItems?: number } = {},
+  ): { items: any[][]; indices: number[][] } {
+    const items: any[][] = [];
+    const indices: number[][] = [];
+    if (!array || !array.length) {
+      return { items, indices };
+    }
+    const { maxSizeInBytes, maxItems } = options;
+    let currentChunk: any[] = [array[0]];
+    let currentIndices: number[] = [0];
+    let currentSize = maxSizeInBytes ? sizeof(array[0]) : 0;
+    for (let idx = 1; idx < array.length; idx++) {
+      const item = array[idx];
+      const itemSizeInBytes = maxSizeInBytes ? sizeof(item) : 0;
+      if (
+        this.isSizeLimitReached(itemSizeInBytes, currentSize, maxSizeInBytes) ||
+        this.isLengthLimitReached(currentChunk.length, maxItems)
+      ) {
+        items.push(currentChunk);
+        indices.push(currentIndices);
+        currentChunk = [];
+        currentIndices = [];
+        currentSize = 0;
+      }
+      currentChunk.push(item);
+      currentIndices.push(idx);
+      currentSize += itemSizeInBytes;
+    }
+
+    if (currentChunk.length > 0) {
+      items.push(currentChunk);
+      indices.push(currentIndices);
+    }
+
+    return { items, indices };
+  }
+
+  static validateBatchResults(input: any[], batchResults: BatchResult[]) {
+    if (!Array.isArray(batchResults)) {
+      throw new BatchError('batch step requires array output from batch executor');
```

Comment:
What will be the structure of BatchStepOutput for this error?

Reply:
whole execution fails if this happens so no need to capture in BatchStepOutput